### PR TITLE
Remove ros arguments

### PR DESCRIPTION
**Pull request by mjcarroll**
Python support for remove command line arguments for ros2/rcl#219

Connects to ros2/rcl#219

### CODE REVIEW

**Comment by mjcarroll** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/4128/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1209/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3411/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4235/)

File: rclpy/test/test_utilities.py [hunk 1]
```diff
@@ -18,6 +18,27 @@
...
+    def test_remove_ros_args(self):
+        args = ['process_name', '-d', '__ns:=/foo/bar', '__ns:=/fiz/buz', '--foo=bar', '--baz']
+        stripped_args = rclpy.utilities.remove_ros_args(args=args)
+        self.assertEqual(4, len(stripped_args))
+        self.assertEqual('process_name', stripped_args[0])
+        self.assertEqual('-d', stripped_args[1])
+        self.assertEqual('--foo=bar', stripped_args[2])
+        self.assertEqual('--baz', stripped_args[3])
+
+    def test_remove_ros_args_empty(self):
```
Comment (mikaelarguedas) [inline rclpy/test/test_utilities.py]: Nit: It seems that this test can be simplified by replacing the try/except block with something like:
```
args = []
with self.assertRaises(RuntimeError):
    rclpy.utilities.remove_ros_args(args=args)
```

Comment (wjwwood) [reply rclpy/test/test_utilities.py]: I was just typing this :)

lgtm otherwise.

Comment (mjcarroll) [reply rclpy/test/test_utilities.py]: Today I learned.

**Comment by mxgrey** [comment]
Regarding the MSVC warning that changed the Windows build to unstable, I believe this is an MSVC front-end parsing error (which occur shockingly often).

It's complaining about implicitly casting a `const char * *` to a `void *` on the premise of [C4090](https://msdn.microsoft.com/en-us/library/k77bkb8d.aspx). The purpose of that warning is to avoid a situation where a pointer that points to a **const** value is given to a pointer that points to a **mutable** value.

However, the lines it's complaining about are **not** converting from pointer-to-const-value to pointer-to-mutable-value. The warning would be correct if we were trying to convert a `char * const *` to a `void *`, but instead we are converting from a `const char * *`, which is equivalent to `char const * *`.

I'm pretty convinced that whatever parsing method is being used to trigger the warning is misreading the cv-qualifications of the pointer.

File: rclpy/src/rclpy/_rclpy.c [hunk 2]
```diff
@@ -221,7 +221,10 @@ rclpy_remove_ros_args(PyObject * Py_UNUSED(self), PyObject * args)
         for (int ii = 0; ii < nonros_argc; ++ii) {
           PyList_SET_ITEM(result_list, ii, PyUnicode_FromString(nonros_argv[ii]));
         }
+#pragma warning(push)
+#pragma warning(disable: 4090)
```
Comment (mxgrey) [inline rclpy/src/rclpy/_rclpy.c]: Maybe post a link to the warning suppression justification in a comment?

https://github.com/ros2/rclpy/pull/180#issuecomment-375452757

Comment (mjcarroll) [reply rclpy/src/rclpy/_rclpy.c]: Put it in the commit message, not in the source, but that's a good idea :)

**Comment by mjcarroll** [comment]
Once more:

 * Linux [Build Status](http://ci.ros2.org/job/ci_linux/4136/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1214/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3416/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4241/)

**Comment by mjcarroll** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/4137/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1215/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3417/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4242/)

File: rclpy/src/rclpy/_rclpy.c [hunk 3]
```diff
@@ -224,10 +224,14 @@ rclpy_remove_ros_args(PyObject * Py_UNUSED(self), PyObject * args)
 /* it was determined that the following warning is likely a front-end parsing issue in MSVC.
  * See: https://github.com/ros2/rclpy/pull/180#issuecomment-375452757
  */
+#if defined(_MSVC_VER)
```
Comment (mikaelarguedas) [inline rclpy/src/rclpy/_rclpy.c]: I think the macro is named `_MSC_VER` (without the V)

**Comment by mjcarroll** [comment]
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/4147/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/1224/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/3428/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/4251/)

**Pull request merged**